src/compiler/tsbuildPublic.ts

```diff
@@ -1324,7 +1347,7 @@ namespace ts {
         buildResult: BuildResultFlags,
         errorType: string,
     ) {
-        const canEmitBuildInfo = !(buildResult & BuildResultFlags.SyntaxErrors) && program && !outFile(program.getCompilerOptions());
+        const canEmitBuildInfo = program && !outFile(program.getCompilerOptions());

```

Comment:
Because we are now saving changeSet in buildInfo, we can emit buildinfo even if there are syntax errors